portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -49,14 +48,19 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
 
     PreparedQuery results = datastore.prepare(q);
     List<Comment> arr = new ArrayList<>();
+    int how_many = Integer.parseInt(request.getParameter("how_many"));
 
     for (Entity it : results.asIterable()) {
       String message = (String) it.getProperty("message");
       String name = (String) it.getProperty("name");
       long timestamp = (long) it.getProperty("timestamp");
       long id = (long) it.getKey().getId();
-      
+
       arr.add(new Comment(message, name, timestamp, id));
+      --how_many;
+    
+      if (how_many == 0)
+        break;
```

Comment:
It can underflow if the number sent in the POST request is <= 0. But this doesn't matter, I coded it in this way to have a nicer approach for the "Display all comments" function from HTML. The for loop terminates when all the comments are put in the list.

Reply:
I see -- please add a comment that explains that.